app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -92,7 +92,6 @@ def ensure_backup_codes_in_session
     end
 
     def generate_codes
-      revoke_remember_device(current_user)
```

Comment:
this looks like it was very intentional, do we know the story of why this was added in the first place?

Reply:
digging in history, there's this old comment

https://github.com/18F/identity-idp/pull/3643/files#diff-26cf7fb1c93eefc7da0ac75e24146e7ba9024b45a25da36abdb22e7d97320fe7L57-L69

>  We don't want to revoke remember device if the user is setting up backup codes as their second MFA

which implies to me, we *do* want to revoke remember device if this is a primary MFA? that comment comes from an earlier PR: https://github.com/18F/identity-idp/pull/3450/files

Reply:
I believe the motivation comes from revoking all remembered devices when deleting MFAs. Backup codes are a little weird since re-generating deletes the old ones if they exist. In this case, we don't know why someone is creating new codes.

Reply:
> I believe the motivation comes from revoking all remembered devices when deleting MFAs. Backup codes are a little weird since re-generating deletes the old ones if they exist. In this case, we don't know why someone is creating new codes.

Would it make sense to keep the revoke, but only call it if the user already had backup codes prior to creating their new ones here?

Reply:
That sounds like a good approach to me,

Reply:
@zachmargolis and @mitchellhenke 
Thank you for the context on that.

Reply:
While it's out of scope for this ticket since we already have an established pattern for revoking remembered device when removing an MFA, I'd wonder if it's worth considering to revoke the cookie _only if the cookie is associated with the MFA being removed_. 

That might require a bigger redesign for how we track remembered device cookies, which may or may not be worthwhile.

Reply:
Should that be a spike ticket?

Reply:
Yeah, it could be. In retrospect, given how infrequently someone is removing an MFA method, I'm not entirely sure it'd be worth the effort to try to optimize. 🤷

Reply:
I think the invalidation of all remembered devices is probably safest/easiest? I'm not sure if it's worthwhile either to try to be that targeted.